src/test/scala/CoreServerTests.scala

```diff
@@ -0,0 +1,340 @@
+import viper.server.ViperBackendConfigs.SiliconConfig
+import viper.server.{AstGenerator, VerificationJobHandler, ViperConfig, ViperCoreServer}
+import viper.silver.logger.{SilentLogger, ViperStdOutLogger}
+import viper.silver.reporter._
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+
+
+class CoreServerTest extends WordSpec with Matchers with ScalatestRouteTest {
+  import scala.language.postfixOps
+
+  implicit var actor_system: ActorSystem = ActorSystem("Test")
+  val client_actor = actor_system.actorOf(ClientActor.props())
+
+  object ClientActor {
+    case object Terminate
+    def props(): Props = Props(new  ClientActor())
+  }
+
+  class ClientActor() extends Actor{
+
+    override def receive: PartialFunction[Any, Unit] = {
+      case _: OverallSuccessMessage =>
+        actor_system.terminate()
+      case _: OverallFailureMessage =>
+        actor_system.terminate()
+    }
+  }
+
+  private val verifiableFile = "src\\test\\resources\\viper\\let.vpr"
+  private val emptyFile ="src\\test\\resources\\viper\\empty.vpr"
+  private val sumFile = "src\\test\\resources\\viper\\sum_method.vpr"
+  private val verificationErrorFile = "src\\test\\resources\\viper\\verification_error.vpr"
+
+  private val console_logger = ViperStdOutLogger("parsingTest logger", "ALL")
+  private val silent_logger = SilentLogger()
+  private val console_reporter = new StdIOReporter()
+  private val silent_reporter = NoopReporter
+
+  "An instance of ViperCoreServer" when {
+    "verifying a single file with caching disabled" should {
+      val file = verificationErrorFile
+      val astGen = new AstGenerator(file, silent_logger)
+
+      val config = new ViperConfig(List())
+      config.verify()
+      var core = new ViperCoreServer(config)
+      "be able to execute 'start()' without exceptions" in {
+        core.start()
+      }
+
+      var handler: VerificationJobHandler = null
+      "be able to execute 'verify()' without exceptions" in {
+        handler = core.verify(file, SiliconConfig(List("--disableCaching")), astGen.viper_ast.get)
+        assert(handler != null)
+      }
+
+      "be able to have 'verifiy()' return a JobHandler with positive id when executed" in {
+        assert(handler.id >= 0)
+      }
+
+      var messages_future: Future[Seq[Message]] = null
+      "be able to have 'getMessagesFuture()' return a future of a sequence of Viper messages." in {
+        messages_future = core.getMessagesFuture(handler.id)
+        assert(messages_future != null)
+      }
+
+      "eventually see the future returned from 'getMessagesFuture()' completed sucessfully" in {
+        while (!messages_future.isCompleted) {
+          Thread.sleep(500)
+        }
+        messages_future.onComplete({
+          case Success(verRes) => succeed
+          case Failure(_) => fail()
+        })
+      }
+
+      "be able to execute 'stop()' without exceptions" in {
+        core.stop()
+      }
```

Comment:
I would like to add a test to see what happens if we ask for something after stop() has been invoked. The expected behaviour should be an exception of appropriate type. This is how we can test that the server can be stopped. Also, there should be some tests that are expected to fail for the entire test suite to be robust.